### ISSUE TITLE
[@typescript-eslint/padded-blocks] Extend base rule

### ISSUE THREAD

**Issue by josephzidell** [typescript-eslint/typescript-eslint#3945]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/padded-blocks": "error"
  }
}
```

```TS
declare namespace A {

    type Foo = 'string';

}
```

**Expected Result**
The padded-blocks rule would catch this

**Actual Result**
The actual result is no error is shown.

**Additional Info**
The [base rule](https://github.com/eslint/eslint/blob/master/lib/rules/padded-blocks.js) checks blocks, classes, and switches. Regular blocks are AST Node type `BlockStatement`, while a declaration block is a `TSModuleBlock`. 

I am willing to create an PR for this. 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.32.0` |
| `@typescript-eslint/parser`        | `4.32.0` |
| `TypeScript`                       | `4.4.3` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `v14.16.1` |

**Comment by bradzacher** [comment]
With any stylistic lint rule - my suggestion is that you should use a tool properly designed to handle formatting like prettier.

Happy to accept a PR if you're so motivated.

**Comment by Josh-Cena** [comment]
Because the ESLint core is also working on [deprecating/removing its formatting rules](https://github.com/eslint/eslint/issues/17522), we will close this. A 3rd-party plugin providing formatting, like https://github.com/eslint-stylistic/eslint-stylistic, can surely support TS OOTB.
